### PR TITLE
refactor helper methods into /utils and /middleware

### PR DESCRIPTION
**Pull request by mattlockyer**
Cleanup and rebase of https://github.com/near/near-contract-helper/pull/152

Eager to make this repo more modular.
This PR refactors a number of methods previously used by routes defined in app.js into their own modules.
Changes to project structure:

/utils
  sms.js
  email.js
/middleware
  account.js
  recovery.js
  2fa.js (wip)

### CODE REVIEW

**Comment by mattlockyer** [comment]
Not expecting this to be accepted yet, there's the e2e test to do with wallet.

Wanted to start clean over on a new branch off latest master.

File: app.js [hunk 1]
```diff
@@ -54,6 +54,25 @@ app.use(async (ctx, next) => {
     ctx.near = await nearPromise;
     await next();
 });
+const SECURITY_CODE_DIGITS = 6;
+const NEW_ACCOUNT_AMOUNT = process.env.NEW_ACCOUNT_AMOUNT;
+const password = require('secure-random-password');
+const models = require('./models');
+/********************************
+Routes
```
Comment (vgrichina) [inline app.js:62]: nit: not clear what this comment tries to explain. Stuff directly below isn't "Routes"

File: app.js [hunk 2]
```diff
@@ -54,6 +54,25 @@ app.use(async (ctx, next) => {
...
+    recoveryMethodsFor,
+    checkRecoveryMethod
+} = require('./middleware/recovery.js');
+const {
+    withAccount,
+    withPublicKey,
+    checkAccountOwnership
+} = require('./middleware/account.js');
 
 const VALID_BLOCK_AGE = 100;
```
Comment (vgrichina) [inline app.js:78]: why `checkAccountOwnership`, `sendSms`, etc not extracted?

**Comment by vgrichina** [comment]
> Not expecting this to be accepted yet, there's the e2e test to do with wallet.

I think easiest way to test it right now (besides tests on CI) would be to change this env var https://github.com/near/near-wallet/blob/master/src/utils/wallet.js#L15 to point wallet to modified contract helper.

**Comment by vgrichina** [comment]
@mattlockyer let's not put new features (like https://github.com/near/near-contract-helper/pull/155/commits/f564b0706839bb88a64aa5133db71cc5c3ac25ea) and refactoring into  the same PR.

Best to keep PRs on the smaller side to make them easier to review and merge.

**Comment by mattlockyer** [comment]
Will break this PR up and resubmit smaller PRs.

**Pull request closed without merging**